Broken macOS build

**Describe the bug**
According to @utrecht 28a9ac6dde0cfb61d2060a5dde1d35c9472a9194 broke the macOS build when using homebrew for installing the dependencies.

Thank you for taking the time to look at this, I do not have a lot of experience with cmake so I could be misunderstanding what is happening.

So working through it from a fresh clone.

I'm on Big Sur 11.2 (20D64), with an intel processor.

libsndfile is installed through homebrew and located at `/usr/local/Cellar/libsndfile/1.0.31 (49 files, 1.8MB) *` with the homebrew created symlinks in `/usr/local/lib` and `/usr/local/include`

Running the following commands from the created `build` directory.

`cmake ..`
`cmake --build .`

Results in building to [100%] and then while attempting to `Linking CXX executable ../../Mumble.app/Contents/MacOS/Mumble` the following error appears.

```Undefined symbols for architecture x86_64:
  "_sf_write_float", referenced from:
      VoiceRecorder::run() in VoiceRecorder.cpp.o
  "_sf_open", referenced from:
      VoiceRecorder::ensureFileIsOpenedFor(SF_INFO&, boost::shared_ptr<VoiceRecorder::RecordInfo>&) in VoiceRecorder.cpp.o
  "_sf_command", referenced from:
      VoiceRecorder::ensureFileIsOpenedFor(SF_INFO&, boost::shared_ptr<VoiceRecorder::RecordInfo>&) in VoiceRecorder.cpp.o
  "_sf_seek", referenced from:
      AudioOutputSample::prepareSampleBuffer(unsigned int) in AudioOutputSample.cpp.o
  "_sf_error", referenced from:
      AudioOutputSample::loadSndfile(QString const&) in AudioOutputSample.cpp.o
      AudioOutputSample::prepareSampleBuffer(unsigned int) in AudioOutputSample.cpp.o
  "_sf_close", referenced from:
      SoundFile::~SoundFile() in AudioOutputSample.cpp.o
      SoundFile::~SoundFile() in AudioOutputSample.cpp.o
      boost::detail::sp_counted_impl_pd<VoiceRecorder::RecordInfo*, boost::detail::sp_ms_deleter<VoiceRecorder::RecordInfo> >::~sp_counted_impl_pd() in VoiceRecorder.cpp.o
      boost::detail::sp_counted_impl_pd<VoiceRecorder::RecordInfo*, boost::detail::sp_ms_deleter<VoiceRecorder::RecordInfo> >::~sp_counted_impl_pd() in VoiceRecorder.cpp.o
      boost::detail::sp_counted_impl_pd<VoiceRecorder::RecordInfo*, boost::detail::sp_ms_deleter<VoiceRecorder::RecordInfo> >::dispose() in VoiceRecorder.cpp.o
  "_sf_set_string", referenced from:
      VoiceRecorder::ensureFileIsOpenedFor(SF_INFO&, boost::shared_ptr<VoiceRecorder::RecordInfo>&) in VoiceRecorder.cpp.o
  "_sf_open_virtual", referenced from:
      SoundFile::SoundFile(QString const&) in AudioOutputSample.cpp.o
  "_sf_read_float", referenced from:
      AudioOutputSample::prepareSampleBuffer(unsigned int) in AudioOutputSample.cpp.o
  "_sf_strerror", referenced from:
      SoundFile::SoundFile(QString const&) in AudioOutputSample.cpp.o
      AudioOutputSample::loadSndfile(QString const&) in AudioOutputSample.cpp.o
      VoiceRecorder::ensureFileIsOpenedFor(SF_INFO&, boost::shared_ptr<VoiceRecorder::RecordInfo>&) in VoiceRecorder.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

If I `export VERBOSE=1` and re-run I see the command that is failing as....

`/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -mmacosx-version-min=10.9 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-dead_strip CMakeFiles/mumble.dir/mumble_autogen/mocs_compilation.cpp.o CMakeFiles/mumble.dir/main.cpp.o CMakeFiles/mumble.dir/About.cpp.o CMakeFiles/mumble.dir/ACLEditor.cpp.o CMakeFiles/mumble.dir/AudioConfigDialog.cpp.o CMakeFiles/mumble.dir/Audio.cpp.o CMakeFiles/mumble.dir/AudioInput.cpp.o CMakeFiles/mumble.dir/AudioOutput.cpp.o CMakeFiles/mumble.dir/AudioOutputSample.cpp.o CMakeFiles/mumble.dir/AudioOutputSpeech.cpp.o CMakeFiles/mumble.dir/AudioOutputUser.cpp.o CMakeFiles/mumble.dir/AudioStats.cpp.o CMakeFiles/mumble.dir/AudioWizard.cpp.o CMakeFiles/mumble.dir/BanEditor.cpp.o CMakeFiles/mumble.dir/CELTCodec.cpp.o CMakeFiles/mumble.dir/Cert.cpp.o CMakeFiles/mumble.dir/ClientUser.cpp.o CMakeFiles/mumble.dir/ConfigDialog.cpp.o CMakeFiles/mumble.dir/ConfigWidget.cpp.o CMakeFiles/mumble.dir/ConnectDialog.cpp.o CMakeFiles/mumble.dir/CustomElements.cpp.o CMakeFiles/mumble.dir/Database.cpp.o CMakeFiles/mumble.dir/DeveloperConsole.cpp.o CMakeFiles/mumble.dir/Global.cpp.o CMakeFiles/mumble.dir/GlobalShortcut.cpp.o CMakeFiles/mumble.dir/GlobalShortcutButtons.cpp.o CMakeFiles/mumble.dir/LCD.cpp.o CMakeFiles/mumble.dir/ListenerLocalVolumeDialog.cpp.o CMakeFiles/mumble.dir/Log.cpp.o CMakeFiles/mumble.dir/LookConfig.cpp.o CMakeFiles/mumble.dir/MainWindow.cpp.o CMakeFiles/mumble.dir/Markdown.cpp.o CMakeFiles/mumble.dir/Messages.cpp.o CMakeFiles/mumble.dir/MumbleApplication.cpp.o CMakeFiles/mumble.dir/NetworkConfig.cpp.o CMakeFiles/mumble.dir/OpusCodec.cpp.o CMakeFiles/mumble.dir/Plugins.cpp.o CMakeFiles/mumble.dir/PTTButtonWidget.cpp.o CMakeFiles/mumble.dir/RichTextEditor.cpp.o CMakeFiles/mumble.dir/Screen.cpp.o CMakeFiles/mumble.dir/ServerHandler.cpp.o CMakeFiles/mumble.dir/Settings.cpp.o CMakeFiles/mumble.dir/SharedMemory.cpp.o CMakeFiles/mumble.dir/SocketRPC.cpp.o CMakeFiles/mumble.dir/SvgIcon.cpp.o CMakeFiles/mumble.dir/TalkingUI.cpp.o CMakeFiles/mumble.dir/TalkingUIContainer.cpp.o CMakeFiles/mumble.dir/TalkingUIEntry.cpp.o CMakeFiles/mumble.dir/TalkingUISelection.cpp.o CMakeFiles/mumble.dir/TextMessage.cpp.o CMakeFiles/mumble.dir/ThemeInfo.cpp.o CMakeFiles/mumble.dir/Themes.cpp.o CMakeFiles/mumble.dir/Tokens.cpp.o CMakeFiles/mumble.dir/Usage.cpp.o CMakeFiles/mumble.dir/UserEdit.cpp.o CMakeFiles/mumble.dir/UserInformation.cpp.o CMakeFiles/mumble.dir/UserListModel.cpp.o CMakeFiles/mumble.dir/UserLocalNicknameDialog.cpp.o CMakeFiles/mumble.dir/UserLocalVolumeDialog.cpp.o CMakeFiles/mumble.dir/UserModel.cpp.o CMakeFiles/mumble.dir/UserView.cpp.o CMakeFiles/mumble.dir/VersionCheck.cpp.o CMakeFiles/mumble.dir/ViewCert.cpp.o CMakeFiles/mumble.dir/VoiceRecorder.cpp.o CMakeFiles/mumble.dir/VoiceRecorderDialog.cpp.o CMakeFiles/mumble.dir/WebFetch.cpp.o CMakeFiles/mumble.dir/XMLTools.cpp.o CMakeFiles/mumble.dir/widgets/CompletablePage.cpp.o CMakeFiles/mumble.dir/widgets/MUComboBox.cpp.o CMakeFiles/mumble.dir/widgets/MultiStyleWidgetWrapper.cpp.o CMakeFiles/mumble.dir/__/ACL.cpp.o CMakeFiles/mumble.dir/__/Channel.cpp.o CMakeFiles/mumble.dir/__/ChannelListener.cpp.o CMakeFiles/mumble.dir/__/Connection.cpp.o CMakeFiles/mumble.dir/__/Group.cpp.o CMakeFiles/mumble.dir/__/__/3rdparty/smallft/smallft.cpp.o CMakeFiles/mumble.dir/SharedMemory_unix.cpp.o CMakeFiles/mumble.dir/AppNap.mm.o CMakeFiles/mumble.dir/GlobalShortcut_macx.mm.o CMakeFiles/mumble.dir/Log_macx.mm.o CMakeFiles/mumble.dir/os_macx.mm.o CMakeFiles/mumble.dir/TextToSpeech_macx.mm.o CMakeFiles/mumble.dir/CrashReporter.cpp.o CMakeFiles/mumble.dir/ManualPlugin.cpp.o CMakeFiles/mumble.dir/Overlay.cpp.o CMakeFiles/mumble.dir/OverlayClient.cpp.o CMakeFiles/mumble.dir/OverlayConfig.cpp.o CMakeFiles/mumble.dir/OverlayEditor.cpp.o CMakeFiles/mumble.dir/OverlayEditorScene.cpp.o CMakeFiles/mumble.dir/OverlayPositionableItem.cpp.o CMakeFiles/mumble.dir/OverlayText.cpp.o CMakeFiles/mumble.dir/OverlayUser.cpp.o CMakeFiles/mumble.dir/OverlayUserGroup.cpp.o CMakeFiles/mumble.dir/PathListWidget.cpp.o CMakeFiles/mumble.dir/Overlay_macx.mm.o CMakeFiles/mumble.dir/Zeroconf.cpp.o CMakeFiles/mumble.dir/__/__/3rdparty/qqbonjour/BonjourServiceBrowser.cpp.o CMakeFiles/mumble.dir/__/__/3rdparty/qqbonjour/BonjourServiceResolver.cpp.o CMakeFiles/mumble.dir/CoreAudio.mm.o CMakeFiles/mumble.dir/JackAudio.cpp.o CMakeFiles/mumble.dir/PAAudio.cpp.o CMakeFiles/mumble.dir/mumble_autogen/EWIEGA46WW/qrc_mumble.cpp.o CMakeFiles/mumble.dir/mumble_autogen/2IKLLRAJFH/qrc_mumble_flags_0.cpp.o CMakeFiles/mumble.dir/mumble_autogen/6RJHKKPRHE/qrc_MumbleTheme.cpp.o CMakeFiles/mumble.dir/mumble_autogen/EJRQKI7XPS/qrc_translations.cpp.o CMakeFiles/mumble.dir/mumble_autogen/6RJHKKPRHE/qrc_ClassicTheme.cpp.o -o ../../Mumble.app/Contents/MacOS/Mumble  ../libshared.a /usr/local/opt/qt5/lib/QtConcurrent.framework/QtConcurrent /usr/local/opt/qt5/lib/QtSql.framework/QtSql /usr/local/opt/qt5/lib/QtSvg.framework/QtSvg /usr/local/opt/qt5/lib/QtWidgets.framework/QtWidgets -framework AppKit -framework ApplicationServices -framework Carbon -framework ScriptingBridge -framework Security /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/lib/libxar.tbd speex/libspeex.a rnnoise/librnnoise.a -framework AudioUnit -framework CoreAudio -framework AVFoundation /usr/local/lib/libprotobuf.dylib /usr/local/opt/openssl/lib/libssl.dylib /usr/local/opt/openssl/lib/libcrypto.dylib /usr/local/opt/qt5/lib/QtNetwork.framework/QtNetwork /usr/local/opt/qt5/lib/QtXml.framework/QtXml -framework CoreServices /usr/local/opt/qt5/lib/QtGui.framework/QtGui /usr/local/opt/qt5/lib/QtCore.framework/QtCore`

If I change to that directory and re-run the failed command and append `/usr/local/Cellar/libsndfile/1.0.31/lib/libsndfile.dylib` the Mumble executable successfully links.

I can also fix it by changing `src/mumble/CMakeLists.txt` to...

```
find_pkg("LibSndFile;sndfile" REQUIRED)

# Look for various targets as they are named differently on different platforms
if(static AND TARGET sndfile-static)
	target_link_libraries(mumble PRIVATE sndfile-static)
elseif(TARGET sndfile)
	target_link_libraries(mumble PRIVATE sndfile)
else()
	target_link_libraries(mumble PRIVATE ${sndfile_LIBRARIES})
endif()```

```
CMake Error at cmake/pkg-utils.cmake:111 (find_package):
  find_package called with invalid argument "sndfile"
Call Stack (most recent call first):
  src/mumble/CMakeLists.txt:353 (find_pkg)
```

--- a/cmake/pkg-utils.cmake
+++ b/cmake/pkg-utils.cmake
@@ -163,7 +163,6 @@ macro(find_pkg ARG_ALIASES)
 		endif()
 	endforeach()
 
-	unset(ALIASES)
 	unset(LAST_ALIAS)
 	unset(FIND_PACKAGE_ARGUMENTS)
 
@@ -182,11 +181,33 @@ macro(find_pkg ARG_ALIASES)
 
 			unset(NAME_UPPER)
 		endif()
+
+		# If multiple capitalizations are searched for this will ensure that the
+		# lowercase name of the library's variables are set as well.
+		# Simplifying usage of find_pkg on case-insensitive file systems.
+		string(TOLOWER ${NAME} NAME_LOWER)
+		if(NOT ${NAME} STREQUAL NAME_LOWER AND ${NAME_LOWER} IN_LIST ALIASES)
+
+			if(NOT ${NAME_LOWER}_VERSION)
+				set(${NAME_LOWER}_VERSION ${${NAME}_VERSION})
+			endif()
+
+			if(NOT ${NAME_LOWER}_LIBRARIES)
+				set(${NAME_LOWER}_LIBRARIES ${${NAME}_LIBRARIES})
+			endif()
+
+			if(NOT ${NAME_LOWER}_FOUND)
+				set(${NAME_LOWER}_FOUND ${${NAME}_FOUND})
+			endif()
+
+		endif()
+		unset(NAME_LOWER)
 	elseif(NOT FIND_PACKAGE_QUIET)
 		if(NOT PkgConfig_FOUND)
 			message(WARNING "pkg-config was not found, consider installing it for better chances of finding a package")
 		endif()
 	endif()
 
+	unset(ALIASES)
 	unset(NAME)
 endmacro()

